### PR TITLE
[iOS] Image wrongly layouted when a transform is applied

### PR DESCRIPTION
**Pull request by carldebilly**
Fix https://github.com/unoplatform/uno/issues/2441

# Bugfix

## What is the current behavior?
Applying a render transform directly on an `<Image>` on iOS were producing a wrongly layouted image.

## What is the new behavior?
Render transform is now applied correctly.

## PR Checklist

- [X] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- ~~[ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ImageTests/Image_Transforms.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Uno.UI.Samples.Controls;
+
+namespace UITests.Windows_UI_Xaml_Controls.ImageTests
+{
+	[SampleControlInfo("Image")]
```
Comment (dr1rrb) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ImageTests/Image_Transforms.xaml.cs]:
```suggestion
	[Sample("Image", "Transform")]
```

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.iOS.cs [hunk 2]
```diff
@@ -287,18 +287,12 @@ public override void LayoutSubviews()
...
 
-		private CGSize _previousSize;
-
-		public override CGSize SizeThatFits(CGSize size)
-		{
-			size = _layouter.Measure(size.ToFoundationSize());
-
-			return _previousSize = size;
-		}
+		public override CGSize SizeThatFits(CGSize size) => _layouter.Measure(size.ToFoundationSize());
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.iOS.cs]: You don't invoke `base.SizeThatFits` ... it might be legit, but...

Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.iOS.cs]: It is.

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 3]
```diff
@@ -60,6 +60,7 @@
 
 ### Bug fixes
 
+- [iOS] Applying a `<RenderTransform>` on an image were producing wrong layout result.
```
Comment (jeromelaban) [inline doc/ReleaseNotes/_ReleaseNotes.md]:
```suggestion
- [iOS] Applying a `<RenderTransform>` on an image was producing an incorrect layout result.
```

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.1

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.1

**Pull request merged**